graphql/schema.lua

```diff
@@ -34,9 +35,101 @@ function schema.create(config, name)
   self:generateTypeMap(introspection.__Schema)
   self:generateDirectiveMap()
 
+  if opts.defaultValues == true then
+    self.defaultValues = {}
+    self.defaultValues.mutation = self:extractDefaults(self.mutation)
+    self.defaultValues.query = self:extractDefaults(self.query)
+  end
+
+  if opts.directivesDefaultValues == true then
+    self.directivesDefaultValues = {}
+
+    for directiveName, directive in pairs(self.directiveMap or {}) do
+      self.directivesDefaultValues[directiveName] = self:extractDefaults(directive)
+    end
+  end
+
   return self
 end
 
+function schema:extractDefaults(node)
+  if not node then return end
+
+  local defaultValues
+  local nodeType = node.__type ~= nil and node or node.kind
+
+  if nodeType.__type == 'NonNull' then
+    return self:extractDefaults(nodeType.ofType)
+  end
+
+  if nodeType.__type == 'Enum' then
+    return node.defaultValue
+  end
+
+  if nodeType.__type == 'Scalar' then
+    return node.defaultValue
+  end
+
+  node.fields = type(node.fields) == 'function' and node.fields() or node.fields
+
+  if nodeType.__type == 'Object' or nodeType.__type == 'InputObject' then
+    for fieldName, field in pairs(nodeType.fields or {}) do
+      local fieldDefaultValue = self:extractDefaults(field)
```

Comment:
So nested structures is supported here? Do there any other issues left for serializing nested structures? (Or I am confusing it with something)

Reply:
self:extractDefaults - recursively called so the answer is Yes